apps/dashboard/app/models/project.rb

```diff
@@ -35,50 +27,57 @@ def find(project_path)
     def dataroot
       Rails.logger.debug("project path is: #{OodAppkit.dataroot.join('projects')}")
 
-      OodAppkit.dataroot.join('projects').tap do |p|
+      OodAppkit.dataroot.join('projects').tap do |path|
+        Rails.logger.debug("tap dataroot running on path = #{path}")
         p.mkpath unless p.exist?
       rescue StandardError => e
         Pathname.new('')
       end
     end
   end
 
-  attr_reader :dir, :description, :icon, :title
+  validates :dir, presence: true
+  validates :dir, format: {
+    with: /[\w-]+\z/,
+    message: 'Name may only contain letters, digits, dashes, and underscores'
+  }
+
+  attr_reader :dir
+  delegate :icon, :title, :description, to: :manifest
 
   def initialize(attributes = {})
     @dir            = attributes.fetch(:dir, nil).to_s
-    @title          = attributes.fetch(:title, nil).to_s
-    @description    = attributes.fetch(:description, nil).to_s
-    @icon           = attributes.fetch(:icon, nil).to_s
   end
 
-  def config_dir
-    Pathname.new("#{absolute_dir}/.ondemand").tap { |p| p.mkpath unless p.exist? }
+  def save!
+    make_manifest
+    write_manifest
```

Comment:
It seems like the difference between `save!` and `save` is whether or not validations raise errors or not.

We should have uniformity between `save!` and `update`.  I think I prefer without the exclamation. It's a little more work to catch the error, but a little cleaner on the controller side.

Which is to say I'm thinking this should just be `save` that actually returns true or false and catches what errors may occur.